### ISSUE TITLE
Workflow stuck due to metadata error - how to fix ?

### ISSUE THREAD

**Issue by cgirardot** [galaxyproject/galaxy#8918]
Hi !

I am in the middle of a WF and I have the following issue 

<img width="797" alt="Screenshot 2019-10-31 at 17 19 12" src="https://user-images.githubusercontent.com/10990795/67965910-35400400-fc03-11e9-8ce5-4a8c64d0a0d3.png">

All dependent jobs are in pause state. Problem is I cannot fix this metadata issue ie :

<img width="1285" alt="Screenshot 2019-10-31 at 17 29 20" src="https://user-images.githubusercontent.com/10990795/67966318-fc545f00-fc03-11e9-8ee8-07102fe17ea0.png">

what is the way out ? Note that I can hit the database if necessary ...everything that is not "restart 
the WF from scratch" is welcome :-)

We are running our local galaxy v 19.05

thx

**Comment by cgirardot** [comment]
Looking into the job info, I can see an error  

```
[main] Version: 0.7.17-r1188
[main] CMD: bwa mem -t 16 -v 1 -A 1 -B 4 -O 6,6 -E 50 -L 0 -U 17 -T 30 -h 5 -5 /g/funcgen/indices/galaxy_bwa_mem/dm6/dm6.fa /g/furlong/incoming/2016-07-04-C8NRHACXX/lane1/LIB21996_RBA21398/fastq/C8NRHACXX_16s003316-1-1_16s004086-1-1_Ghavi-helm_lane116s004086_2_sequence.txt.gz
[main] Real time: 2577.995 sec; CPU: 30042.244 sec
[bam_sort_core] merging from 48 files and 16 in-memory blocks...

[E::bgzf_read] Read block operation failed with error 33 after 0 of 4 bytes
```
If I re-run the job, the new job is not replacing the currently staled one in the collection. 

So I guess my question becomes:
- how can I force re-run the job within the collection; or 
- how can I set these jobs to failed state; galaxy would then offer me to rerun the failed job in place.

**Comment by cgirardot** [comment]
ok here is how (with admin rights) I could fix this : 
1. Delete the link child job to be able to re-run the "metadata auto-detection". 
2. This did not work most likely due to the error I described above. So I re-run the job and replace the dataset file of the stuck job with the result of the new run (a simple cp)
3. Click the  "metadata auto-detection" on the stuck jobs, this worked
4. Undelete the child job with the hope that the handler would then pick this up, but it did not
5. Tried to directly `UPDATE job SET state = 'paused' where id =XXX` but this did not turned the job into a paused one oh the UI (why ?). When clicking the 're-run' job button the UI did not offer the "resume paused dependencies" option that I need
6. Tried to directly `UPDATE job SET state = 'error' where id =XXX` and this worked : When clicking the 're-run' job button the UI DID offer the "resume paused dependencies". It is now running and should now work🤞

To avoid users to get stuck as I was, I think a simple solution like re-run with the "resume paused dependencies" should be offered on job which failed at the metadata detection.
